src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -66,166 +61,222 @@ public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand,
+                "cover.advanced_energy_detector.mode_eu", "cover.advanced_energy_detector.mode_percent")
+                .setTooltipHoverString("cover.advanced_energy_detector.modes_tooltip")
         );
 
-        // get/set min EU
-        group.addWidget(new LabelWidget(10, 5 + 3 * (SIZE + PADDING), "cover.advanced_energy_detector.min"));
-        group.addWidget(new ImageWidget(68, 3 * (SIZE + PADDING), 4 * SIZE, SIZE, GuiTextures.DISPLAY));
-        group.addWidget(new TextFieldWidget2(72, 5 + 3 * (SIZE + PADDING), 4 * SIZE, SIZE,
-                this::getMinValue, this::setMinValue)
-                    .setMaxLength(19)
-                    .setAllowedChars(ALLOWED_CHARS)
-                    .setPostFix(" EU")
+        // invert logic button
+        group.addWidget(new LabelWidget(10, 5 + 4 * (SIZE + PADDING), "cover.advanced_energy_detector.invert_label"));
+        group.addWidget(new CycleButtonWidget(72, 4 * (SIZE + PADDING), 4 * SIZE, SIZE, this::isInverted, this::setInverted,
+                "cover.advanced_energy_detector.normal", "cover.advanced_energy_detector.inverted")
+                .setTooltipHoverString("cover.advanced_energy_detector.invert_tooltip")
         );
 
-        return ModularUI.builder(GuiTextures.BACKGROUND, 176, 108)
+        return ModularUI.builder(GuiTextures.BACKGROUND, 176 + (3 * SIZE), 108 + (SIZE))
                 .widget(group)
+                .widget(widgetsToUpdate) // add synced widgets
                 .build(this, player);
     }
 
+    private WidgetGroup constructWidgetsToUpdate() {
+        WidgetGroup sync = new WidgetGroup();
+        sync.addWidget(new TextFieldWidget2(76, 5 + (SIZE + PADDING), 8 * SIZE, SIZE, this::getMinValue, this::setMinValue)
+                .setMaxLength(19)
+                .setAllowedChars(TextFieldWidget2.NATURAL_NUMS)
+                .setPostFix(this.getPostFix()));
+        sync.addWidget(new TextFieldWidget2(76, 5 + 2 * (SIZE + PADDING), 8 * SIZE, SIZE, this::getMaxValue, this::setMaxValue)
+                .setMaxLength(19)
+                .setAllowedChars(TextFieldWidget2.NATURAL_NUMS)
+                .setPostFix(this.getPostFix()));
+        return sync;
+    }
+
     private String getMinValue() {
-        return String.valueOf(minEU);
+        return String.valueOf(minValue);
     }
 
     private String getMaxValue() {
-        return String.valueOf(maxEU);
+        return String.valueOf(maxValue);
     }
 
     private void setMinValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            this.minEU = Math.min(maxEU - 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.minEU = Math.max(maxEU - 1, 0);
+        long parsedValue;
+
+        if (usePercent) {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MIN_PERCENT);
+        } else {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MIN_EU);
         }
```

Comment:
You could write this as `parsedValue = GTUtility.tryParseLong(val, usePercent ? DEFAULT_MIN_PERCENT : DEFAULT_MIN_EU);` which will make it more readable

Reply:
resolved with https://github.com/GregTechCEu/GregTech/pull/1511/commits/cf30e48f846acd7ceb072986679853d079006628